### DISCUSSION TITLE
Add support for freeform reactions

### DISCUSSION THREAD

**Discussion by kittykat** [element-hq/element-meta#1795] [Product Feedback]
#### What would you like to do?

I would like to be able to send freeform reactions on all platforms

#### Why would you like to do it?

- [MSC2677](https://github.com/uhoreg/matrix-doc/blob/aggregations-reactions/proposals/2677-reactions.md) supports reactions with long keys, i.e. text reactions
- This feature is already available through other apps and our users can see those reactions
- It is already possible to send freeform reactions through /devtools and bots from Element, so at the moment the feature exists but is obscure
- Element users can already "+1" a freeform reaction on any platform by clicking it

#### How would you like to achieve it?

- By being able to compose a freeform reaction on any platform
- Being able to view long reactions in a user friendly way \(I can't see any upper limit set by the MSC\)
- Be able to [report reactions to moderators](https://github.com/matrix-org/matrix-react-sdk/pull/6628#issuecomment-1123509348)
- Remove reactions as a moderator/admin user in a room

#### Sub-issues

- \[ \] https://github.com/vector-im/element-web/issues/19409
- \[ \] vector-im/element-android#6028
- \[ \] vector-im/element-ios#6147

**Comment by JokerGermany** [comment]
Would be cool if it would support displaying pictures as reaction as well.
e.G. this would be useful for Discord Bridges.

**Comment by 10Meisterbaelle** [comment]
I don't understand why this is not a feature in the official Matrix app. It's litterally a Matrix feature.

**Comment by daniellekirkwood** [comment]
This is not something we're open to at the minute so I'm going to go ahead and make it a discussion
